src/controller/CommissionerDiscoveryController.cpp

```diff
@@ -606,10 +627,23 @@ void CommissionerDiscoveryController::CommissioningSucceeded(uint16_t vendorId,
     mVendorId  = vendorId;
     mProductId = productId;
     mNodeId    = nodeId;
+
+    auto rotatingId = std::string{};
+    auto passcode = uint32_t{ 0 };
+
+    auto client = GetUDCClientState();
+    if (client != nullptr)
+    {
+        // Get rotating ID and cached (commissioner?) passcode to handle AccountLogin
+        rotatingId = getRotatingIdString(client);
```

Comment:
Since we always need rotating id at the end of commissioning now, feel free to cache it locally in the CommissionerDiscoveryController object (adjacent to mPasscode), so that you can cleanup some duplicate code in this file.

The first time we enter this flow is OnUserDirectedCommissioningRequest and so that's probably the best place to cache it.

Reply:
Updated to cache rotating ID in OnUserDirectedCommissioningRequest and use that value when calling HandleLogin, as suggested. Kept the other calculations of rotating ID as is just to be safe, in case the cache might be stale (?).